packages/core/test/actions.test.ts

```diff
@@ -3178,6 +3178,49 @@ describe('sendTo', () => {
       ]
     `);
   });
+
+  it('a self-event "handler" sent using sendTo should be able to read updated snapshot of self', () => {
+    const spy = jest.fn();
+    const machine = createMachine({
+      context: {
+        counter: 0
+      },
+      initial: 'a',
+      states: {
+        a: {
+          on: { NEXT: 'b' }
+        },
+        b: {
+          entry: [
+            assign({ counter: 1 }),
+            sendTo(({ self }) => self, { type: 'EVENT' })
+          ],
+          on: {
+            EVENT: {
+              actions: ({ self }) => spy(self.getSnapshot().context),
+              target: 'c'
+            }
+          }
+        },
+        c: {}
+      }
+    });
+
+    const actorRef = createActor(machine).start();
+
+    actorRef.send({ type: 'NEXT' });
+    actorRef.send({ type: 'EVENT' });
+
+    expect(spy).toMatchMockCallsInlineSnapshot(`
+[
+  [
+    {
+      "counter": 1,
+    },
+  ],
+]
+`);
```

Comment:
I just pushed out this test case that shows, what I believe to be, a regression. Note that I pushed out 2 test cases here: https://github.com/statelyai/xstate/pull/4954/commits/ece423d68c449eeb0ae5dee8392e4e10a989ce63

They both are identical, the only difference is that one uses `raise(...)` whereas the other one uses `sendTo(({ self }) => self, ...)`. While the raise one doesn't make a lot of sense, it clearly shows that something is broken here because those 2 tests are exactly the same if we consider `raise` to be a sugar for the other one.